classes/media/AudioHandler.php

```diff
@@ -236,4 +236,33 @@ public function getLongDesc($file) {
 
 		return wfMessage('ev_audio_long_desc', strtoupper($extension), $stream->getCodecName(), $wgLang->formatTimePeriod($format->getDuration()), $wgLang->formatBitrate($format->getBitRate()))->text();
 	}
+
+	/**
+	 * Runs FFProbe and caches results in the Main WAN Object cache
+	 *
+	 * @param string|FSFile|File $file The file to work on
+	 * @param string $select Video / Audio track to select
+	 * @return array
+	 */
+	protected function getFFProbeResult( $file, string $select = 'v:0' ): array {
+		$path = $file;
+
+		if ( $file instanceof \File || $file instanceof \FSFile ) {
+			$path = $file->getPath();
+		}
+
+		if ( $file === false ) {
```

Comment:
Could you clarify when `$file` could be `false`? The DocBlock indicates that it should always be a `string`, `FSFile`, or `File`. If it doesn't meet these types, shouldn't we avoid calling this method altogether? Alternatively, we could return early in the method. Executing all the code above seems redundant if the type is incorrect.